### PR TITLE
feat: make command add subdirectories support

### PR DESCRIPTION
**Pull request by DevHaoZi**
Closes goravel/goravel#72

## 📑 Description
make command add subdirectories support

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [x] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

## ℹ Additional Information
none.

### CODE REVIEW

File: auth/console/policy_make_command.go [hunk 1]
```diff
@@ -57,16 +57,48 @@ func (receiver *PolicyMakeCommand) getStub() string {
...
+	name = strings.TrimSuffix(name, ".go")
+
+	segments := strings.Split(name, "/")
+
+	policyName := segments[len(segments)-1]
+
+	packageName := "policies"
+
+	if len(segments) > 1 {
+		packageName = strings.Join(segments[:len(segments)-1], "/")
```
Comment (hwbrzzl) [inline auth/console/policy_make_command.go:94]: Q: Should be `packageName = segments[len(segments)-2]`?

File: auth/console/policy_make_command_test.go [hunk 2]
```diff
@@ -20,5 +20,11 @@ func TestEventMakeCommand(t *testing.T) {
 	err = policyMakeCommand.Handle(mockContext)
 	assert.Nil(t, err)
 	assert.True(t, file.Exists("app/policies/user_policy.go"))
+
+	mockContext.On("Argument", 0).Return("User/AuthPolicy").Once()
+	err = policyMakeCommand.Handle(mockContext)
+	assert.Nil(t, err)
+	assert.True(t, file.Exists("app/policies/User/auth_policy.go"))
```
Comment (hwbrzzl) [inline auth/console/policy_make_command_test.go:27]: S: Add assert: `assert.True(file.Contain("app/policies/User/auth_policy.go", "package User"))` and `assert.True(file.Contain("app/policies/User/auth_policy.go", "type AuthPolicy struct {"))`

File: console/console/make_command.go [hunk 3]
```diff
@@ -49,14 +49,48 @@ func (receiver *MakeCommand) getStub() string {
...
+	name = strings.TrimSuffix(name, ".go")
+
+	segments := strings.Split(name, "/")
+
+	commandName := segments[len(segments)-1]
+
+	packageName := "commands"
+
+	if len(segments) > 1 {
+		packageName = strings.Join(segments[:len(segments)-1], "/")
```
Comment (hwbrzzl) [inline console/console/make_command.go:86]: Ditto and below

File: console/console/make_command_test.go [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+func TestMakeCommand(t *testing.T) {
+	makeCommand := &MakeCommand{}
+	mockContext := &consolemocks.Context{}
+	mockContext.On("Argument", 0).Return("CleanCache").Once()
+	assert.Nil(t, makeCommand.Handle(mockContext))
+	assert.True(t, file.Exists("app/console/commands/clean_cache.go"))
+
+	mockContext.On("Argument", 0).Return("Goravel/CleanCache").Once()
+	assert.Nil(t, makeCommand.Handle(mockContext))
+	assert.True(t, file.Exists("app/console/commands/Goravel/clean_cache.go"))
```
Comment (hwbrzzl) [inline console/console/make_command_test.go:21]: Ditto and below

**Review by hwbrzzl — COMMENTED** [review]
Great!

File: auth/console/policy_make_command.go [hunk 5]
```diff
@@ -57,16 +58,40 @@ func (receiver *PolicyMakeCommand) getStub() string {
...
+
+	segments := strings.Split(name, "/")
+
+	policyName := segments[len(segments)-1]
+
+	packageName := "policies"
+	folderPath := ""
+
+	if len(segments) > 1 {
+		folderPath = strings.Join(segments[:len(segments)-1], "/")
```
Comment (hwbrzzl) [inline auth/console/policy_make_command.go:92]: Q: Also need to use `filepath.Join`?

File: support/str/str.go [hunk 6]
```diff
@@ -2,24 +2,24 @@ package str
...
-	"math/rand"
+	"crypto/rand"
 	"strconv"
 	"strings"
 	"unicode"
-
-	supporttime "github.com/goravel/framework/support/time"
 )
 
 func Random(length int) string {
```
Comment (hwbrzzl) [inline support/str/str.go:11]: S: Could you please add a test case for `length == 0`? I forgot to add it before。

Comment (DevHaoZi) [reply support/str/str.go:11]: > S: Could you please add a test case for `length == 0`? I forgot to add it before。

ok

File: support/str/str_test.go [hunk 7]
```diff
@@ -10,6 +10,10 @@ func TestRandom(t *testing.T) {
 	assert.Len(t, Random(10), 10)
 }
 
+func TestZeroRandom(t *testing.T) {
+	assert.Equal(t, "", Random(0))
+}
```
Comment (hwbrzzl) [inline support/str/str_test.go:15]:
```
func TestRandom(t *testing.T) {
        assert.Len(t, Random(10), 10)
	assert.Empty(t, Random(0))
}
```

File: event/console/event_make_command.go
Comment (hwbrzzl) [inline event/console/event_make_command.go:1]: Add a test unit to `event/console/event_make_command_test.go`

File: event/console/listener_make_command.go
Comment (hwbrzzl) [inline event/console/listener_make_command.go:1]: Add a test unit to `event/console/listener_make_command_test.go`

**Review by hwbrzzl — COMMENTED** [review]
Just found another two tips ~

**Review by hwbrzzl — APPROVED** [review]
Perfect! LGTM

**Pull request merged**